src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts

```diff
@@ -160,26 +167,19 @@ export class DotEditContentHtmlService {
      * @memberof DotEditContentHtmlService
      */
     renderAddedContentlet(contentlet: DotPageContent): void {
-        const doc = this.getEditPageDocument();
-        const containerEl = doc.querySelector(
-            // tslint:disable-next-line:max-line-length
-            `div[data-dot-object="container"][data-dot-identifier="${this.currentContainer.identifier}"][data-dot-uuid="${
-                this.currentContainer.uuid
-            }"]`
-        );
+        this.renderAddedItem({
+            item: contentlet,
+            checkExistFunc: this.isContentExistInContainer.bind(this),
+            getContent: this.dotContainerContentletService.getContentletToContainer.bind(this.dotContainerContentletService)
+        });
+    }
 
-        if (this.isContentExistInContainer(contentlet, containerEl)) {
-            this.showContentAlreadyAddedError();
-        } else {
-            const contentletEl: HTMLElement = this.createNewContentlet(contentlet);
-            containerEl.insertAdjacentElement('beforeend', contentletEl);
-            this.dotContainerContentletService
-                .getContentletToContainer(this.currentContainer, contentlet)
-                .subscribe((contentletHtml: string) => {
-                    this.renderHTMLToContentlet(contentletEl, contentletHtml);
-                    this.currentAction = DotContentletAction.EDIT;
-                });
-        }
+    renderAddedForm(form: ContentType): void  {
+        this.renderAddedItem({
+            item: form,
+            checkExistFunc: this.isFormExistInContainer.bind(this),
```

Comment:
Why? why not just call `this.isFormExistInContainer` directly? why pass that around?

Reply:
because this method is call twice each one with it own 'checkExistFunc'